apps/communications/dialer/js/call_log_db.js

```diff
@@ -1336,121 +1392,138 @@ var CallLogDBManager = {
           cursor.update(group);
           count++;
           cursor.continue();
-        } else {
-          self._asyncReturn(callback, count);
-          self._updateCacheRevision();
         }
       };
-      req.onerror = function onerror(event) {
-        self._asyncReturn(callback, event.target.error.name);
-      };
     });
   },
   invalidateContactsCache: function invalidateContactsCache(callback) {
     var self = this;
```

Comment:
This is a pretty big function and wondering if there is a way to break it up so that it's easier to follow.